server/src/main/java/org/elasticsearch/index/mapper/ValueFetcherSource.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.query.SearchExecutionContext;
+
+import java.util.List;
+
+/**
+ * Resolves a {@link ValueFetcher} for a particular field.
+ */
+public interface ValueFetcherSource {
```

Comment:
having source in this name, given that value fetchers may retrieve fields from _source, is super confusing :)

Reply:
Fair! Got any better ideas?